@xen-orchestra/lite/src/components/Tree.tsx

```diff
@@ -96,45 +106,82 @@ const CustomContent = React.forwardRef(function CustomContent(props: CustomConte
   )
 })
 
-const renderItem = ({ children, id, label, to, tooltip }: ItemType) => {
+const renderItem = ({ children, id, label, to, tooltip }: ItemType, defaultSelectedNode?: string) => {
   return (
     <TreeItem
       ContentComponent={CustomContent}
       // FIXME: ContentProps should only be React.HTMLAttributes<HTMLElement> or undefined, it doesn't support other type.
       // when https://github.com/mui-org/material-ui/issues/28668 is fixed, remove 'as CustomContentProps'.
-      ContentProps={{ to } as CustomContentProps}
+      ContentProps={{ defaultSelectedNode, to } as CustomContentProps}
       label={tooltip ? <Tooltip title={tooltip}>{label}</Tooltip> : label}
       key={id}
       nodeId={id}
     >
-      {Array.isArray(children) ? children.map(renderItem) : null}
+      {Array.isArray(children) ? children.map(item => renderItem(item, defaultSelectedNode)) : null}
     </TreeItem>
   )
 }
 
 const Tree = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
   {
-    initialState: ({ defaultSelectedNodes }) => ({
-      selectedNodes: defaultSelectedNodes === undefined ? [] : defaultSelectedNodes,
-    }),
+    initialState: ({ collection, defaultSelectedNodes }) => {
+      if (defaultSelectedNodes === undefined) {
+        return {
+          expandedNodes: [collection[0].id],
+          selectedNodes: [],
+        }
+      }
+
+      // expandedNodes should contain all nodes up to the defaultSelectedNodes.
+      const expandedNodes: Array<string> = []
+      const tempExpandedNodes: Set<string> = new Set()
```

Comment:
To avoid `Set` repetition, you can do this:
```suggestion
      const tempExpandedNodes = new Set<string>()
```